### ISSUE TITLE
Make Unified English Braille the default Braille code

### ISSUE THREAD

**Issue by codeofdusk** [nvaccess/nvda#6952]
Since UEB has now been adopted as the international standard for English Braille, it should be the default Braille table for English users.

**Comment by codeofdusk** [comment]
Oh forget it.
Here's the patch (somebody can merge this in manually).
[0001-Fix-6952.patch.txt](https://github.com/nvaccess/nvda/files/832154/0001-Fix-6952.patch.txt)

**Comment by jcsteh** [comment]
We definitely shouldn't use grade 2 by default, since some users may not be
able to read that. I think grade 1 probably makes sense, though.

This should probably be done as part of #2439, as that work completely
changes the braille table list.

**Comment by dkager** [comment]
> We definitely shouldn't use grade 2 by default, since some users may not be able to read that. I think grade 1 probably makes sense, though.

Agreed. Especially if this is the default table every users ends up with after installation. Also, why does the patch not set the default input table to UEB? That makes much mroe sense IMO.

**Comment by jcsteh** [comment]
UEB input isn't possible until #2439 is done (which isn't even yet in
master).

**Comment by leonardder** [comment]
I'm leaning towards a vote against this. English computer braille has been the standard table for years and years. Where UEB might be a default for English speaking countries, this is not the case for other countries where the English computer code is used often, such as the netherlands.

**Comment by jcsteh** [comment]
The problem with this argument is that, at least from my experience,
English computer braille is not exactly something that is taught when one
learns braille. Also, I don't think I've seen lower numbers anywhere
outside of a computing or mathematical context. Finally, this is
specifically U.S. computer braille; there is also U.K. computer braille
which is different again.

**Comment by leonardder** [comment]
How about allowing language maintainers to set their preferred default 
braille tables? I think I suggested this earlier, but there was no real 
use case for it then.

**Comment by dkager** [comment]
The current US computer braille table sets a bad first impression. I can see why you don't want that as a default. Uncontracted UEB sounds good, assuming the rules for capital signs aren't too complex to be picked up by those who aren't educated in UEB.

> How about allowing language maintainers to set their preferred default braille tables?

In The Netherlands the US computer braille table (extended version, #6836) is commonly used. But then, so is the German computer braille table. So what would you choose as the language-specific default?

**Comment by leonardder** [comment]
How about allowing language maintainers to set their preferred default 
braille tables?
>
> In The Netherlands the US computer braille table (extended version, 
> #6836 <https://github.com/nvaccess/nvda/issues/6836>) is commonly 
> used. But then, so is the German computer braille table. So what would 
> you choose as the language-specific default?
>
Well, if UEB becomes standard for English, I don't see a reason not to 
choose the Dutch table here.

**Comment by dkager** [comment]
It would maybe make more sense to use the Dutch 6-dot table as default, since that is actually standardized.
Liblouis is also working on table metadata. Part of that is the language, presumably in some standard format. NVDA could then map the operating system language to a braille table. Of course users need to be able to override this with their own choice.

**Comment by ruifontes** [comment]
I agree with Leonard...
It is better to have a default Braille table for each market...

Rui


-----Mensagem Original----- 
De: Leonard de Ruijter
Data: 11 de março de 2017 10:25
Para: nvaccess/nvda
Cc: Subscribed
Assunto: Re: [nvaccess/nvda] Make Unified English Braille the default 
Braille code (#6952)

How about allowing language maintainers to set their preferred default
braille tables?
>
> In The Netherlands the US computer braille table (extended version,
> #6836 <https://github.com/nvaccess/nvda/issues/6836>) is commonly
> used. But then, so is the German computer braille table. So what would
> you choose as the language-specific default?
>
Well, if UEB becomes standard for English, I don't see a reason not to
choose the Dutch table here.

**Comment by leonardder** [comment]
< Liblouis is also working on table metadata. Part of that is the language, presumably in some standard format. NVDA could then map the operating system language to a braille table. Of course users need to be able to override this with their own choice.

I think this is a very good point to implement. We can do this in a similar way nvda's language is selected in espeaks initialization.

**Comment by jcsteh** [comment]
I feel there's some serious scope confusion/bloat happening here. :)

Having language specific default tables is definitely ideal, but is a much larger chunk of work than just changing the default. I don't think the liblouis metadata is going to be sufficient for this (even once it is finally ready). Unless I'm missing something, yes, it will specify the language of a table, but there are already multiple tables per language, so that isn't helpful. We need to be able to choose a single default. Regardless, this should be addressed separately.

As for changing the default we have right now, if non-English languages really feel U.S. computer braille is a better default than UEB grade 1, I think we should drop this idea altogether until we have language specific defaults. I'd like to see solid arguments as to why U.S. computer braille is better for beginning users, though; see https://github.com/nvaccess/nvda/issues/6952#issuecomment-285836305 for what I don't think this is the case. Note that @dkager seems happy enough with UEB grade 1 as the default; see https://github.com/nvaccess/nvda/issues/6952#issuecomment-285857535.

**Comment by dkager** [comment]
In looking through open braille issues I found this one again.

> As for changing the default we have right now, if non-English languages really feel U.S. computer braille is a better default than UEB grade 1, I think we should drop this idea altogether until we have language specific defaults. I'd like to see solid arguments as to why U.S. computer braille is better for beginning users, though; [...]

One reason why U.S. computer braille is easier (not better as such) is that it has very basic rules for capitalization and numbers: they have their own symbols. For example, `UEB` has a different capital sign in Unified English Braille than `Ueb`. In fact, the capital sign isn't standardized across non-English languages, so somebody is bound to be unfamiliar with the UEB approach. Furthermore, signs like the quote (") change based on context. For example, in `"quote (") quote"` I see 3 different braille signs for this symbol when using UEB.

The flip side of this is that the number sign, period and comma do match Dutch literary braille, and probably many others. But they aren't standardized either. For example, in Finnish braille the UEB period (⠲) corresponds to the exclamation mark.

Long story short: you can't win. But maybe we can decide on a default table now that we have discussed it for a while. I would propose UEB grade 1 inspite of the caveats above. I say that mostly because UEB goes beyond the U.S. and therefore will be familiar to a large user base. Probably the largest single user base of any table in liblouis.

**Comment by leonardder** [comment]
I just discovered that there is a ticket (#290) for language based braille table defaults.
